<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<link href="default.css" type="text/css" rel="stylesheet" />
<script language='javascript' type='text/javascript'>
function update_modtime() {
  var x = new Date(document.lastModified);
  document.getElementById("last_mod").innerHTML="Last modified: "+x;
}
</script>
<title>Biochemistry Research - Pearson</title>
</head>
<body bgcolor='white' onload='update_modtime();'>

<img align='right' src="wrp_crop_2018.jpg" width=150/>
<h2>William R. Pearson</h2>
<i>Professor of Biochemistry</i><br />
Ph.D., 1977, California Institute of Technology
<p><a href="mailto:wrp@virginia.edu">wrp@virginia.edu</a>
<p>
Dept. of Biochemistry and Molecular Genetics<br />
Box 800733<br />
Charlottesville, VA 22908<br />
<p>
 FAX: (434)-924-5069
<p>
<a href="https://orcid.org/0000-0002-0727-3680" target="orcid.widget" rel="noopener noreferrer" style="vertical-align:top;"><img src="https://orcid.org/sites/default/files/images/orcid_16x16.png" style="width:1em;margin-right:.5em;" alt="ORCID iD icon">orcid.org/0000-0002-0727-3680</a>
<hr />
<h3><a href="mol_evol/">Course Similarity Searching workshop</a>
<h3><a href="biol4230/">Biol4230 -- Bioinformatics and Functional Genomics</a>
<hr />
<h3><a href="https://github.com/wrpearson/fasta36/">The FASTA sequence comparison package source code</a></h3>
<h3><a href="http://fasta.bioch.virginia.edu/fasta_www2/">The FASTA WWW search page</a></h3>
<hr />
<p>
The <a href="http://fasta.bioch.virginia.edu/">FASTA</a>
programs can be used to search protein and DNA sequence databases, and
to confirm the statistical significance of a match by comparing the
alignment score to a distribution of scores produced by shuffled
sequences.  Programs are also available to display local alignments.
<p>
The FASTA programs extend the BLAST local similarity programs by
offering rigorously optimal local and global alignments, a much wider
range of alignment scoring matrices
(<a href="#scoring_matrix">Pearson, 2013)</a>, and translated
alignments that allow frame-shifts, and alignments with oligo-peptides
and oligo-nucleotides.  In addition, the FASTA programs can highlight
functional residues in alignments, and provide sub-alignment scoring
to reduce alignment over-extension (<a href="#overextend">Mills and
Pearson, 2013</a>, <a href="#psiover">Gonzalez and Pearson, 2010</a>).
<hr />
<h3>PROTEIN EVOLUTION</h3>
<p>
We have a long-standing interest in exploiting protein sequence
information, both for understanding better how new protein sequences
arise and for understanding the relationship between protein sequence
and protein structure.  Since the description of the FASTP program in
1985, our group has been developing more effective methods for
identifying distantly related protein sequences.  Over the past 10
years, state-of-the-art methods have improved to where proteins that
have diverged from a common ancestor in the past billion years are
likely to be detected by sequence similarity searching. We hope to
push back that threshold to beyond 2 billion years (near the time when
prokaryotes and eukaryotes diverged), but already it is possible to
identify novel proteins that are likely to have emerged in the last
500 - 800 million years.  If we can identify proteins that emerged in
the last 100 - 250 million years, it may be possible to identify the
mechanisms by which new proteins are formed.
<hr />
<p>
<a href="papers/ismb2000.pdf">ISMB 2000 Tutorial on Protein Evolution and Protein Sequence Comparison</a> (PDF file)
<hr />
<h3>Selected Publications</h3>
<a name="articles" />
<p>
Insana, G., Martin, M. J., Pearson, W. R. (2024) "Improved selection of canonical proteins for reference proteomes" NAR Genom Bioinform. 10.1093/nargab/lqae066 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11165316/"/>PMCID: PMC11165316</a> <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11165316/pdf/lqae066.pdf" target=pdf>[PDF]</a>
<p>
Triant, D. A., Pearson, W. R. (2022) "Comparison of detection methods and genome quality when quantifying nuclear mitochondrial insertions in vertebrate genomes" Front Genet. 13:984513 doi: 10.3389/fgene.2022.984513 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9723244/">PMCID: PMC9723244</a>  <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9723244/pdf/fgene-13-984513.pdf" target=pdf>[PDF]</a>
<p>
Pearson, W. R., Li, W., and Lopez, R. (2017) "Query-seeded iterative similarity searching improves selectivity 5&mdash;20-fold" Nuc. Acids Res. 10.1093/nar/gkw1207 <a href="https://academic.oup.com/nar/article-pdf/45/7/e46/25365225/gkw1207.pdf" target=pdf>[PDF]</a>.
<p>
Triant, D. A. and Pearson, W. R. (2015) "Most partial domains in proteins are  alignment and annotation artifacts" Genome Biology 16:99
<a href="http://www.ncbi.nlm.nih.gov/pubmed/25976240" target=entry>[Entrez]</a>
<a href="http://www.genomebiology.com/2015/16/1/99" target=paper>[Journal]</a>
<a href="http://www.genomebiology.com/content/pdf/s13059-015-0656-7.pdf" target=pdf>[PDF]</a>
<p>
Furnham N, Holliday GL, de Beer TA, Jacobsen JO, Pearson WR, Thornton JM. (2013) "The Catalytic Site Atlas 2.0: cataloging catalytic sites and residues identified in enzymes." Nucleic Acids Res. 42:D485-489
<a href="http://www.ncbi.nlm.nih.gov/pubmed/24319146">[Entrez]</a>
<a href="http://nar.oxfordjournals.org/content/42/D1/D485.full.pdf">[PDF]</a>"
<p>
<a name="overextend" />
Mills, L. J. and Pearson, W. R. (2013)
"Adjusting Scoring Matrices to Correct Overextended Alignments"
Bioinformatics. 29:3007-2013 doi: 10.1093/bioinformatics/btt517.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/23995390">[Entrez]</a>
<a href="http://bioinformatics.oxfordjournals.org/cgi/reprint/btt517">[PDF]</a>
<p>
<a name="scoring_matrix" />
Pearson, W. R. (2013)
Curr. Prot. Bioinformatics Chapter 3: Unit 3.5 "Selecting the Right Similarity-Scoring Matrix"  doi: 10.1002/0471250953.bi0305s43.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/24509513">[Entrez]</a>
<a href="papers/wrp_cpbio_0305.pdf">[PDF]</a>
<p>
Pearson, W. R. (2013) "An Introduction to Similarity ("Homology") Searching"
Curr. Prot. Bioinformatics Chapter 3: Unit 3.1 doi: 10.1002/0471250953.bi0301s42. <a href="http://www.ncbi.nlm.nih.gov/pubmed/23749753">[Entrez]</a>
<a href="papers/wrp_cpbio_0301.pdf">[PDF]</a>
<p>
Li W, McWilliam H, Goujon M, Cowley A, Lopez R, Pearson WR.  (2012)
"PSI-Search: iterative HOE-reduced profile SSEARCH searching."
Bioinformatics. <a 
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=22539666">[Entrez]</a>
<a href="http://bioinformatics.oxfordjournals.org/content/early/2012/04/24/bioinformatics.bts240.full.pdf">[PDF]</a>
<p>
Holliday GL, Andreini C, Fischer JD, Rahman SA, Almonacid DE, Williams ST, Pearson WR. (2012) "MACiE: exploring the diversity of biochemical reactions."
Nucleic Acids Res. 2012 Jan;40(Database issue):D783-9.
<a 
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=22058127">[Entrez]</a>
<a href="http://nar.oxfordjournals.org/content/40/D1/D783.full.pdf+html">[PDF]</a>
<p>
M. W. Gonzalez and W. R. Pearson (2010) Bioinformatics "RefProtDom: A protein database with improved domain boundaries and homology relationships" 26:2361-2361 <a href="http://www.ncbi.nlm.nih.gov/pubmed/20693322">[Entrez]</a>
<a href="http://bioinformatics.oxfordjournals.org/cgi/reprint/btq426v1?maxtoshow=&amp;hits=10&amp;RESULTFORMAT=&amp;fulltext=pearson&amp;searchid=1&amp;FIRSTINDEX=0&amp;resourcetype=HWCIT">[PDF]</a>
<p>
M. L. Sierk, M. E. Smoot, E. J. Bass, and W. R. Pearson (2010)
"Improving pairwise sequence alignment accuracy using near-optimal alignments"
BMC Bioinformatics 11:146 doi:10.1186/1471-2105-11-146 <a 
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=20307279">[Entrez]</a>
<a href="http://www.biomedcentral.com/1471-2105/11/146">[PDF]</a>

<p>
<a name='psiover' />
M.  W. Gonzalez and W. R. Pearson (2010) "Homologous over-extension: a challenge for iterative similarity searches" Nuc. Acids Research 38:2177-2189 <a 
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=20064877">[Entrez]</a>
<a href="http://nar.oxfordjournals.org/cgi/reprint/gkp1219?ijkey=egrUX94QhypNuzI&amp;keytype=ref">[PDF]</a>

<p>

D. T. Lavelle and W. R. Pearson (2010) "Globally, unrelated protein sequences appear random" Bioinformatics 26:310-318 <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=19948773">[Entrez]</a> <a href="papers/bioinfo10_wrp_dtl.pdf">[PDF]</a>

<p>
W. R. Pearson and M. L.  Sierk (2005) "The limits of protein sequence
comparison?" Curr Opin Struct Biol. 15:254-260. <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=15919194">[Entrez]</a> <a href="papers/sierk_wrp_currop05.pdf">[PDF]</a>
<p>
W. R. Pearson and T. C. Wood (2001) "Statistical significance in
biological sequence comparison" in Handbook of Statistical Genetics,
D. J. Balding, M. Bishop, and C. Cannings eds. London: Wiley, pp.
39-65
<p>
T. C. Wood and W. R. Pearson
Evolution of Protein Sequences and Structures (1999)
J.Mol. Biol. 291:977-995
<p>
Pearson, W. R., (1998)
Empirical statistical estimates for sequence similarity scores
J.Mol. Biol. 276:71-84
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=98179551">[Entrez]</a>.
<hr />
Ivarsson Y, Mackey AJ, Edalat M, Pearson WR, Mannervik B. (2003)
"Identification of residues in glutathione transferase capable of driving functional diversification in evolution. A novel approach to protein redesign."
J Biol Chem. 2003 278:8733-8738. 
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=12486119">[Entrez]</a>.
<p>
Xu, S.-j., Wang, Y.-p., Roe, B., Pearson, W. R. (1998)
Characterization of the Human Class Mu Glutathione S-Transferase Gene
Cluster and the GSTM1 Deletion.  J. Biol. Chem. 273:3517-3527.  <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=9452477">[Entrez]</a>
<p>
Pearson, W. R., Vorachek, W. R., Xu, S., Berger, R., Hart,
I., Vannais, D., and Patterson, D. (1993)  Identification of
class-mu glutathione transferase genes GSTM1 - GSTM5 on
human chromosome 1p13. Am. J. Human Genet. 53:220-233.
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=93304417">[Entrez]</a>
<p>
Daly, A. K., Thomas, D. J., Cooper, J., Pearson, W. R.,
Neal, D. E., and Idle, J. R. (1993)  Homozygous deletion of
the glutathione S-transferase M1 (GSTM1) gene is a risk
factor in bladder cancer. Brit. Med. J. 307:481-482.
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=PubMed&amp;cmd=Retrieve&amp;dopt=Citation&amp;list_uids=94004140">[Entrez]</a>
<hr />
<h3><a href="papers/">Reprints from the Pearson Lab</a></h3>
<address>
<a href="mailto:wrp@virginia.edu">wrp@virginia.edu</a><br />
</address>
<hr />
<p id="last_mod"></p>
</body>
</html>
