<html>
<head>
<!-- <meta http-equiv="Refresh" content="300"> -->
<link href="default.css" type="text/css" rel="stylesheet">
<!-- updated June, 2013 for Honey bee searches -->
<script type="text/javascript">
<!-- 
    var default_show ='block';

    function set_block() {
       if (window.location.search) {
          default_show ='block';
       }
       else default_show ='block';
    }

    function show_hide(id) {
       var e = document.getElementById(id);
       if(e.style.display == 'block') { e.style.display = 'none';}
       else {
          e.style.display = default_show;
       }
    }
//-->
<!-- Programming for Biology - Similarity Searching Exercises -->
<!-- Computational Genomics 2012 - Similarity Searching Exercises -->
<!-- EBI Bioinformatics Summer School 2016 -->
<!-- BIMS6000 - Similarity searching exercises -->
<!-- Molecular Evolution - Similarity Searching Exercises -->
<!-- Computational Genomics 2014 - Similarity Searching Exercises -->
</script>
<style>
  .box {display: inline-block; width: 20px; height: 9px; margin: 1px;}
  span.c_0 { background-color:slategrey;}
  span.c_1 { background-color:lightgreen;}
  span.c_2 { background-color:lightblue;}
  span.c_3 { background-color:pink;}
  span.c_4 { background-color:cyan;}
  span.c_5 { background-color:tan;}
  span.c_6 { background-color:gold;}
  span.c_7 { background-color:plum;}
  span.c_8 { background-color:mediumplum;}
</style>
<!-- EBI Bioinformatics Summer School 2017 -->
<!-- Molecular Evolution - Similarity Searching Exercises -->
<title>
Programming For Biology 2025 - Similarity Searching
</title>
</head>
<body bgcolor=white onload="set_block()">
<!-- <h1><a href="http://tinyurl.com/fasta-work">tinyurl.com/fasta-work</a></h1> -->
<h1>
<a href="http://fasta.bioch.virginia.edu/mol_evol/">fasta.bioch.virginia.edu/mol_evol</a>&nbsp;&nbsp;&nbsp;&nbsp;
<a href="http://tinyurl.com/fasta-demo">tinyurl.com/fasta-demo</a></h1>
<!--
BIMS6000 --  2020 - Similarity Searching Exercises
-->
<h2>
Programming for Biology --  2025 - Similarity Searching Exercises
</h2>
<p>
<hr />
<p>
These exercises use programs on the <a
href="http://fasta.bioch.virginia.edu/fasta_www2/fasta_list2.shtml"
target="fasta_win">FASTA WWW Search page</a> and the 
<!-- Computational Genomics -->
<a href="http://fasta.bioch.virginia.edu/fasta_www3/fasta_www.cgi?rm=blast"
target='blast_win'>BLAST WWW Search page</a>.
<p>
In the links
below, <a href="http://fasta.bioch.virginia.edu/fasta_www2/" target='fasta_win'>[pgm]</a>
indicates a link with most of the information filled in; e.g. the
program name, query, and
library.  <a href="http://ncbi.nlm.nih.gov/"  target='entrez_win'>[seq]</a> links go to the
NCBI, for more information about the sequence.  In general, you should
click <a>[pgm]</a> links, but not <a>[seq]</a> links.
<p>
<hr>
<b> Identifying homologs and non-homologs; effects of scoring matrices and algorithms; using domain annotations</b>
<hr>
<p>
Use the <a
href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=select&pgm=fa&query=295842263&db=p&annot_seq2=5"
target="fasta_win">FASTA search page [pgm]</a> to compare Honey bee
glutathione transferase D1 NP_001171499/ H9KLY5_APIME <a href="http://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?db=protein&list_uids=295842263&dopt=fasta" TARGET=entrez_win>[seq]</a> (gi|295842263) to the PIR1 Annotated protein sequence database.  Be sure to press <input type='submit' value='Search Database' />, not <input type='submit' value='Align two sequences'>.  
<ol>
<li>Take a look at the output.
<ol type='a'>
<p>
<li>
How long is the query sequence? 
<li> How many sequences are in the PIR1 database?
<li> What scoring matrix was used?
<li> What were the gap penalties? (what is the penalty for a one-residue gap? two residues? -- this is a "trick" question)
<li> What are each of the numbers after the description of the library sequence?  Which one is best for inferring homology?
<li> How similar is the highest scoring sequence?  What is the difference between %_id and %_sim? Why is there no 100% identity match?
<li> Looking at an alignment, where are the boundaries of the alignment (the best local region)?  How many gaps are in the best alignment?  The second best?
</ol>
<p>
<li>
Homologs, non-homologs, and the statistical control.
<ol type='a'>
<p>
<li>
What is the highest (worst) E()-value shown?  What <u>should</u> the
highest (worst) E()-value calculated in the search be (approximately)?
<li>
Which alignment has the worst statistically significant (E()<0.001) score?  Do you think this sequence is likely to be homologous?
<li>
What is the highest scoring (most significant) non-homolog? (The non-homolog with the
highest alignment score, or the lowest E()-value.)  Why do you think it is not homologous?
Look for positive evidence (e.g. a non-homologous domain) for non-homology.
<p>
You can use the domain diagrams (colors) to identify distant homologs, and, by elimination, the highest scoring non-homolog. You can also use the <tt><a href="http://www.uniprot.org/">Sequence&nbsp;Lookup</a></tt> link to Uniprot to look at Domains and Families.
<li>
If the
statistical estimates are accurate, what <u>should</u> the E()-value for the
highest non-homolog (the highest score by chance) be?  (This is a
control for statistical accuracy.)
<li>
What is the E()-value of the most distant homolog shown (based on displayed domain content)?  Could there be more distant <b>homologs</b> in the database?
<li>
How would you confirm that your candidate non-homolog was truly
unrelated?  (<i>Hint</i> - compare your candidate non-homolog
with <b>SwissProt</b> or <b>QFO78/Uniprot Ref</b> for a more comprehensive test.</i>)
</ol>
<p>
<li>
Domains and alignment regions
<ol type='a'>
<p>
<li>
There are three parts to the domain display, the domain structure of
the query (top) sequence (if available), the domain structure of the library (bottom)
sequence, and the domain alignment boundaries in the middle (inside the
alignment box).  The boundaries and color of the alignment domain
coloring match the <tt>Region:</tt> sub-alignment scores.  
<li>
Note that the alignment of Honey bee <tt>GSTD1</tt>
and <tt>SSPA_ECO57</tt> includes portions of both the N-terminal and
C-terminal domains, but neither domain is completely aligned. Why might
the alignments not include the complete domains?
<li>
Is your explanation for the partial domain alignment consistent the
the belief that domains have a characteristic length because they represent compact folding units? How might you
test whether a complete domain is present?
</ol>
<p>
<li>
Note that this Honey bee glutathione transferase shares significant
similarity with both sequences from bacteria (SSPA_ECO57, stringent
starvation protein) and mammals.  How might you test whether the
stringent starvation protein is homologous to glutathione
transferases? (<i>Hint - compare your candidate non-homolog
with <b>SwissProt</b> for a more comprehensive test.</i>)
<p>
<li>
In a new window, repeat the
<a href="http://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=select&pgm=fa&query=295842263&db=p&annot_seq1=4&annot_seq2=4&hide_align=1"
target="fasta_win">GSTD1 search [pgm]</a> using the BLASTP62/-11/-1
scoring matrix <select><option>BlastP62 (30%)</option></select> that
BLAST uses.<br />
Re-examine the honey bee/<tt>SSPA_ECO57</tt> alignment and compare the alignment that you got with BLOSUM50 (the first search) with the alignment you get with BLASTP62.
<pp>
  (In the afternoon lecture and workshop, we will discuss how
  different scoring matrices have different preferred evolutionary
  distances.  BLOSUM50 is more sensitive for alignmets that are about
  25% identical, while BLOSUM62 works best for 30% identical
  alignments.  Keep that in mind while answering the following
  questions.)
<ol type='a'>
<p>
<li>
Do both Glutathione transferase domains align in the honey bee
protein??  Does that imply that one of the domains is not present?
<pp>
Look at the alignments to the homologs above and
below <tt>SSPA_ECO57</tt>.  Based on those.  aligments, do you think
the Glutathione-S-Trfase C-like domain is really missing from the
honey bee protein?
<li>
Why did the alignment become shorter?
<li>
Why might a domain appear to be present in the first (BLOSUM50) search, but not in the second (BLOSUM62)?
</ol>
<li>
Do the same Honey bee GSTD1 search (<tt>295842263</tt>) using the Course
<a
href="http://fasta.bioch.virginia.edu/fasta_www3/fasta_www.cgi?rm=blast&query=295842263&db=p"
target="blast_win"> BLAST [pgm]</a> WWW page.
<ol type='a'>
<li>Take a look at the output.
<ol type="i">
<li>
How long is the query sequence? 
<li> How many sequences are in the PIR1 database?
<li> What scoring matrix was used?
<li> What were the gap penalties?
<li> What are the numbers after the description of the library sequence?  Which one is best for inferring homology?
<li> Looking at an alignment, where are the boundaries of the alignment (the best local region)?
</ol>
<p>
<li>
What is the highest scoring non-homolog?
<p>
<li> How do the <b>BLASTP</b> E()-values compare with the
<b>FASTA</b> (<tt>BLASTP62</tt>) E()-values for the distantly related
mammalian and plant sequences?
<p>
</ol>
<li>
  One of the slides this afternoon will show an alignment between
  human and bovin PCBP2 (Poly(rC) binding protein 2).  One of the
  missing regions of the human protein is from residues 267-279 in
  PCB2_HUMAN (Q15366) when aligning A0A3Q1LX34_BOVIN.
  <pp>
<ol type='a'>
<pp>
<li>
  Use the <a href="https://fasta.bioch.virginia.edu/fasta_www3/fasta_www.cgi?rm=compare&pgm=fa&query=PCB2_HUMAN&q_type=acc&query2=A0A3Q1LX34_BOVIN&q2_type=acc">align two sequences [pgm]</a> link from the <a href=https://fasta.bioch.virginia.edu/fasta_www3/>FASTA web site</a> to align <tt>PCB2_HUMAN</tt> (enter into top box) with <tt>A0A3Q1LX34_BOVIN</tt> (lower box).
<pp>
  Note the locations (coordinates) of the two gaps in <tt>PCB2_HUMAN</tt>.
<li>
    Go to the <a href="https://genome.ucsc.edu/cgi-bin/hgTracks?db=bosTau9&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr5%3A26543511%2D26562251&hgsid=3269020979_57XlW4DjpRupHUVF3LA96EHPfuXh" target='ucsc'>UCSC Genome web site</a> to look at cow PCB2 gene.  These coordinates display the cow genomic sequence from Chr5:26,543,511-26,562,251 (from the end of exon 10 to the beginning of exon 11)
<pp>

 Use the <b>View</b> menu option to get the DNA sequence of the cow PCB2 gene.  First select <tt>View/DNA Sequence</tt>, then click the <tt>Get DNA</tt> option at the bottom of the page.
<pp>
Copy the DNA sequence, and use the   Use the <a href="https://fasta.bioch.virginia.edu/fasta_www3/fasta_www.cgi?rm=compare&pgm=tfx&query=PCB2_HUMAN&q_type=acc">align two sequences [pgm]</a> link  <tt>PCB2_HUMAN</tt> with <tt>cow PCB2 DNA sequence</tt>, that you copied/pasted from the UCSC <tt>Get DNA</tt> link.
<li>
    Use TFASTX to compare residues 267-289 of PCB2_HUMAN to the DNA
    sequence from the cow PCBP2 gene by pasting in the cow genomic sequence from Chr5:26543511:26562251
    (from the end of exon 10 to the beginning of exon 11):
<pp>
Compare the subset range of PCBP2_HUMAN 267-279 to the cow genomic
region using "Compare sequences" with TFASTX <a href='https://fasta.bioch.virginia.edu/fasta_www3/fasta_www.cgi?rm=compare&pgm=tfx&query=PCBP2_HUMAN&start=267&stop=279&annot_seq1=14' target=gene_comp>[pgm]</a> using the "VT20" and
"BLASTP62" matrices.  How long are the alignments? The percent
identity?  What the bit scores? What are the E()-values of the two
alignments?
<pp>
Try the same comparison with residues 169-198 of PCB2_HUMAN
using <tt>VT20</tt> to align exon 7 of <tt>PCB2_HUMAN</tt> to the cow
genome region.  Can you find the missing 4 amino acids at 169-172?
</ol>
<p>
Where to get the FASTA package: 
<a href="http://github.com/wrpearson/fasta36">github.com/wrpearson/fasta36<a>
<p>
The "normal" <a href="http://fasta.bioch.virginia.edu/">FASTA WWW site:</a>
<p>
Contact Bill Pearson: <a href="mailto:wrp@virginia.edu">wrp@virginia.edu</a>
<p>
<hr>
</body>
</html>
