<html><head><title>FASTA command line options</title></head><body bgcolor=white><h3>FASTA command line options</h3><table border=1 cellspacing=0 cellpadding=2> <tr>  <td>-a</td>  <td>  Show entire length of  both sequences in alignment (fasta, ssearch)  </td> </tr> <tr>  <td>  -A  </td>  <td>  Do full Smith-Waterman  alignment for DNA (fasta)  </td> </tr> <tr>  <td>  -b #  </td>  <td>  Number of high scores  to display  </td> </tr> <tr>  <td>  -C #  </td>  <td>  Length of sequence name  labeling aligned sequences  </td> </tr> <tr>  <td>  -d #  </td>  <td>  Number of alignments to  display  </td> </tr> <tr>  <td>  -D  </td>  <td>  Use DEBUG mode  </td> </tr> <tr>  <td>  -E #  </td>  <td>  Expectation threshold  for displaying scores, alignments  </td> </tr> <tr>  <td>  -f #  </td>  <td>  Gap-open penalty (-10  default for proteins with default BLOSUM50 matrix)  </td> </tr> <tr>  <td>  -F #  </td>  <td>  Expectation lower limit  for displaying scores, alignments  </td> </tr> <tr>  <td>  -g #  </td>  <td>  Gap-extension penalty  (-2 default for proteins with default BLOSUM50 matrix)  </td> </tr> <tr>  <td>  -h #  </td>  <td>  Frameshift between  codon penalty (fastax, fasty, tfastx, tfasty)  </td> </tr> <tr>  <td>  -H  </td>  <td>  Omit histogram (fasta  programs); include histogram (pvcomp/mpcomp programs)  </td> </tr> <tr>  <td>  -i  </td>  <td>  Search with reverse  complement query only (fasta DNA, fastx, fasty)  </td> </tr> <tr>  <td>  -j #  </td>  <td>  Frameshift within codon  penalty (fasty, tfasty)  </td> </tr> <tr>  <td>  -J #:#  </td>  <td>  First query, last query  (pvcomp/mpcomp)  </td> </tr> <tr>  <td>  -K #  </td>  <td>  Node buffer size  (pvcomp/mpcomp)  </td> </tr> <tr>  <td>  -l file  </td>  <td>  FASTLIBS library  </td> </tr> <tr>  <td>  -L  </td>  <td>  Long sequence  descriptions on alignments  </td> </tr> <tr>  <td>  -m #  </td>  <td>  Alignment display  options  </td> </tr> <tr>  <td>  -M #:#  </td>  <td>  Library length range;  only sequences within the residue range are considered  </td> </tr> <tr>  <td> -n </td>  <td> Query sequence is DNA </td> </tr> <tr>  <td> -N # </td>  <td> Library segment length   (examine long DNA sequences in overlapping lengths of # nt)  </td> </tr> <tr>  <td> -o </td>  <td> Turn-off band  optimization (fasta) </td> </tr> <tr>  <td> -O file </td>  <td> Output file name (most useful with MacOS9) </td> </tr> <tr>  <td>  -p  </td>  <td> Query sequence is protein </td> </tr> <tr>  <td> -q/-Q </td>  <td> Quiet - do not prompt for any input </td> </tr> <tr>  <td> -r #/# </td>  <td> DNA match/mismatch  scores (+5/-4 default) </td> </tr> <tr>  <td> -R file </td>  <td> Score results file (for debugging) </td> </tr> <tr>  <td> -s file </td>  <td>Scoring matrix: file name or BL50, BL62, BL80, P10, P20,P40,P120, P250 </td> </tr> <tr>  <td>  -S  </td>  <td> Treat lower case as low complexity </td> </tr> <tr>  <td>  -t #  </td>  <td>  NCBI genetic code translation table  </td> </tr> <tr>  <td>  -T # </td>  <td> Number of workers (pvcomp/mpcomp, threaded) </td> </tr> <tr>  <td> -v </td>  <td> Shuffle window size </td> </tr> <tr>  <td> -w </td>  <td>  Alignment display width (default 60)  </td> </tr> <tr>  <td>  -W  </td>  <td> Alignment context (unaligned residues shown around alignment, default=30, fasta,ssearch)  </td> </tr> <tr>  <td> -x # </td>  <td> Score for match to 'X' (protein) or 'N' (DNA), default -1 </td> </tr> <tr>  <td> -X #,# </td>  <td>virtual coordinate of query sequence, library sequences </td> </tr> <tr>  <td>  -z #  </td>  <td>  Statistical estimation  strategy  </td> </tr> <tr>  <td>  -Z #  </td>  <td>  Effective database size  </td> </tr> <tr>  <td>  -1  </td>  <td>  Sort results by <i>init1</i> (best single ungapped region)  </td> </tr> <tr>  <td>  -3  </td>  <td>  Search with forward strand only  </td> </tr></table></body></html>