<html>
<head>
<!-- <meta http-equiv="Refresh" content="300"> -->
<link href="default.css" type="text/css" rel="stylesheet">
<!-- updated June, 2013 for Honey bee searches -->
<script type="text/javascript">
<!-- 
    var default_show ='block';

    function set_block() {
       if (window.location.search) {
          default_show ='block';
       }
       else default_show ='block';
    }

    function show_hide(id) {
       var e = document.getElementById(id);
       if(e.style.display == 'block') { e.style.display = 'none';}
       else {
          e.style.display = default_show;
       }
    }
//-->
<!-- Programming for Biology - Similarity Searching Exercises -->
<!-- Computational Genomics 2012 - Similarity Searching Exercises -->
<!-- EBI Bioinformatics Summer School 2016 -->
<!-- BIMS6000 - Similarity searching exercises -->
<!-- Molecular Evolution - Similarity Searching Exercises -->
<!-- Computational Genomics 2014 - Similarity Searching Exercises -->
</script>
<style>
  .box {display: inline-block; width: 20px; height: 9px; margin: 1px;}
  span.c_0 { background-color:slategrey;}
  span.c_1 { background-color:lightgreen;}
  span.c_2 { background-color:lightblue;}
  span.c_3 { background-color:pink;}
  span.c_4 { background-color:cyan;}
  span.c_5 { background-color:tan;}
  span.c_6 { background-color:gold;}
  span.c_7 { background-color:plum;}
  span.c_8 { background-color:mediumplum;}
</style>
<!-- EBI Bioinformatics Summer School 2017 -->
<!-- Molecular Evolution - Similarity Searching Exercises -->
<title>
Computational Genomics 2023 - Similarity Searching Exercises
</title>
</head>
<body bgcolor=white onload="set_block()">
<!-- <h1><a href="http://tinyurl.com/fasta-work">tinyurl.com/fasta-work</a></h1> -->
<h1>
<a href="https://fasta.bioch.virginia.edu/mol_evol/">fasta.bioch.virginia.edu/mol_evol</a>&nbsp;&nbsp;&nbsp;&nbsp;
<a href="http://tinyurl.com/fasta-demo">tinyurl.com/fasta-demo</a></h1>
<!--
BIMS6000 --  2020 - Similarity Searching Exercises
-->
<h2>
Computational Genomics ECG2023 -- Similarity Searching Exercises
</h2>
<p>
<hr />
<p>
These exercises use programs on the <a
href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_list2.shtml"
target="fasta_win">FASTA WWW Search page</a> and the 
<!-- Computational Genomics -->
<a href="https://fasta.bioch.virginia.edu/fasta_www3/fasta_www.cgi?rm=blast_plus"
target='blast_win'>BLAST WWW Search page</a>.
<p>
In the links
below, <a href="https://fasta.bioch.virginia.edu/fasta_www2/" target='fasta_win'>[pgm]</a>
indicates a link with most of the information filled in; e.g. the
program name, query, and
library.  <a href="https://ncbi.nlm.nih.gov/"  target='entrez_win'>[seq]</a> links go to the
NCBI, for more information about the sequence.  In general, you should
click <a>[pgm]</a> links, but not <a>[seq]</a> links.
<p>
<hr>
<b> Identifying homologs and non-homologs; effects of scoring matrices and algorithms; using domain annotations</b>
<hr>
<p>
<p>
Use the <a
href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=select&pgm=fa&query=295842263&db=p&annot_seq2=5"
target="fasta_win">FASTA search page [pgm]</a> to compare Honey bee
glutathione transferase D1 NP_001171499/ H9KLY5_APIME <a href="https://www.ncbi.nlm.nih.gov/entrez/viewer.fcgi?db=protein&list_uids=295842263&dopt=fasta" TARGET=entrez_win>[seq]</a> (gi|295842263) to the PIR1 Annotated protein sequence database.  Be sure to press <input type='submit' value='Search Database' />, not <input type='submit' value='Align two sequences'>.  
<p>
<ol>
<li>Take a look at the output.
<ol type='a'>
<p>
<li>
How long is the query sequence? 
<li> How many sequences are in the PIR1 database?
<li> What scoring matrix was used?
<li> What were the gap penalties? (what is the penalty for a one-residue gap? two residues? -- this is a "trick" question)
<li> What are each of the numbers after the description of the library sequence?  Which one is best for inferring homology?
<li> How similar is the highest scoring sequence?  What is the difference between %_id and %_sim? Why is there no 100% identity match?
<li> Looking at an alignment, where are the boundaries of the alignment (the best local region)?  How many gaps are in the best alignment?  The second best?
</ol>
<p>
<li>
Homologs, non-homologs, and the statistical control.
<ol type='a'>
<p>
<li>
What is the highest (worst) E()-value shown?  What <u>should</u> the
highest (worst) E()-value calculated in the search be (approximately)?
<li>
Which alignment has the worst statistically significant (E()<0.001) score?  Do you think this sequence is likely to be homologous?
<li>
What is the highest scoring (most significant) non-homolog? (The non-homolog with the
highest alignment score, or the lowest E()-value.)  Why do you think it is not homologous?
Look for positive evidence (e.g. a non-homologous domain) for non-homology.
<p>
You can use the domain diagrams (colors) to identify distant homologs, and, by elimination, the highest scoring non-homolog. You can also use the <tt><a href="https://www.uniprot.org/">Sequence&nbsp;Lookup</a></tt> link to Uniprot to look at Domains and Families.
<li>
If the
statistical estimates are accurate, what <u>should</u> the E()-value for the
highest non-homolog (the highest score by chance) be?  (This is a
control for statistical accuracy.)
<li>
What is the E()-value of the most distant homolog shown (based on displayed domain content)?  Could there be more distant <b>homologs</b> in the database?
<p>How would you confirm that your candidate non-homolog was truly
unrelated?
<ol type='i'>
<li>The highest scoring sequence that does not have <tt>glutathione
transferase</tt> in the description is <tt>SSPA_ECO57</tt>.  How can
we be certain that this is really Glutathione transferase homolog? Do
a search
with <a href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=select&query=P0ACA5&q_type=acc&db=Protein&lib=A&pgm=fa">SSPA_ECO57
[pgm]</a> against a larger sequence database, e.g. <b>SwissProt</b>
or <b>QFO78/Uniprot Ref</b>, for a more comprehensive perspective.
<p>
Do you think <tt>SSPA_ECOLI</tt> is a glutathione transferase homolog?
<li>
To find the highest scoring <b>non-homolog</b>, 
go down the list of high scoring sequences search those sequences against <b>SwissProt</b>.
<p>
Based on those searches, what is the highest scoring non-homolog?  If I tell you that <tt>EF1D2_ARATH</tt> might contain an unannotated glutatione transferase domain, what would be the next highest scoring non-homolog?  Confirm this with a search against <b>SwissProt</b>.
<li>
There are several proteins with "glutathione transferase" in their descriptions that do not have significant scores against the honey bee protein.  Why not?  How would you show that they are probably glutathione transferase homologs?
</ol>
</ol>
<p>
<li>
Domains and alignment regions
<ol type='a'>
<p>
<li>
There are three parts to the domain display, the domain structure of
the query (top) sequence (if available), the domain structure of the library (bottom)
sequence, and the domain alignment boundaries in the middle (inside the
alignment box).  The boundaries and color of the alignment domain
coloring match the <tt>Region:</tt> sub-alignment scores.  
<li>
Note that the alignment of Honey bee <tt>GSTD1</tt>
and <tt>SSPA_ECO57</tt> includes portions of both the N-terminal and
C-terminal domains, but neither domain is completely aligned. Why do
you think the alignments do not include the complete domains?
<li>
Is your explanation for the partial domain alignment consistent the
the argument that domains have a characteristic length? How might you
test whether a complete domain is present?
</ol>
<p>
<li>
Repeat the
<a href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=select&pgm=fa&query=295842263&db=p&annot_seq1=4&annot_seq2=4&hide_align=1"
target="fasta_win">GSTD1 search [pgm]</a> using the BLASTP62/-11/-1
scoring matrix <select><option>BlastP62 (30%)</option></select> that
BLAST uses.<br />
Re-examine the honey bee/<tt>SSPA_ECO57</tt> alignment.
<ol type='a'>
<p>
<li>
Are both Glutathione transferase domains present in the honey bee protein??
<li>
Look at the alignments to the homologs above and below <tt>SSPA_ECO57</tt>.
Based on those.
aligments, do you think the Glutathione-S-Trfase C-like domain is
really missing from the honey bee protein? 
<li>
Why did the alignment become shorter?
<li>
Why would a domain appear to be present in the first (BLOSUM50) search, but not in the second (BLOSUM62)?
</ol>
<p>
<li>
Do the same Honey bee GSTD1 search (<tt>295842263</tt>) using the Course
<a
href="https://fasta.bioch.virginia.edu/fasta_www3/fasta_www.cgi?rm=blast&query=295842263&db=p"
target="blast_win"> BLAST [pgm]</a> WWW page (turn on Pfam Annotation on the PIR1 database)
<ol type='a'>
<p>
<li>Take a look at the output.
<ol type="i">
<li>
How long is the query sequence? 
<li> How many sequences are in the PIR1 database?
<li> What scoring matrix was used?
<li> What were the gap penalties?
<li> What are the numbers after the description of the library sequence?  Which one is best for inferring homology?
<li> Looking at an alignment, where are the boundaries of the alignment (the best local region)?
</ol>
<p>
<li>
What is the highest scoring non-homolog?
<p>
<li> How do the <b>BLASTP</b> E()-values compare with the
<b>FASTA</b> (<tt>BLASTP62</tt>) E()-values for the distantly related
mammalian and plant sequences?
</ol>
<p>
<hr />
<li>Exploring domains with local alignments --- Calmodulin
<ol type='a'>
<li> Use <a href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=lplalign&pgm=lpal&query=calm1_human&q_type=acc&query2=calm1_human&q2_type=acc"
target="fasta_win">lalign/plalign [pgm]</a> to examine local similarities between calmodulin <tt>CALM1_HUMAN</tt> and itself.

<li>Trace the top scoring (non-identical alignment) on the right along the plot of the alignment on  the left.
<li>How many domains are aligned in the top-scoring alignment?
<li>
Do the same <a href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=lplalign&pgm=lpal&query=calm1_human&q_type=acc&query2=calm1_human&q2_type=acc">alignment [pgm]</a>, but check the "annotate sequence" boxes to highlight the annotated EF-hand domains.
<li>
What happens to the domain alignment plot when you use the VT40 scoring matrix.  Do you see the same sets of alignments with VT40?  With shallow scoring matrices, are the domains fully aligned?
<li>
What parts of the domains align with VT20?
</ol>
<p>
<li>Exploring domains with local alignments --- Cortactin (SRC8_HUMAN)
<ol type="a">
<li> Use <a href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=lplalign&pgm=lpal&query=src8_human&q_type=acc&query2=src8_human&q2_type=acc"
target="fasta_win">lalign/plalign&nbsp;[pgm]</a> to examine local
similarities between <tt>SRC8_HUMAN</tt> and itself.  Check the options to "annotate sequence 1 domains" and "annotate sequence 2 domains".
<ol type='i'>
<li>
Identify the third highest scoring alignment on both the alignment plot and the text alignment.
<li>
What is the percent identity of the alignment?
<li>
How many domains are aligned in this alignment?
<li>
Estimate the fraction identical in the first 20 residues of the alignment.  In the last 100 residues.
<li>
Do the ends of the alignment correspond to the domain boundaries?
</ol>
<p>
<li>
Based on the percent identities you saw in the sequence alignment (a), what would the appropriate scoring matrix be to accurately identify the cortactin domains?
<ol type='i'>
<li>
Using a "correct" scoring matrix, the third highest scoring alignment on both the alignment plot and the text alignment.
<li>
What is the percent identity of the alignment (did you pick the right matrix?)
<li>
How many domains are aligned?
<li>
Do the ends of the alignment correspond to the domain boundaries?
</ol>
<p>
<li>
Try some other scoring matrices.  What happens to the relationship between domain ends and alignment ends when the scoring matrix is too deep (BP62)?  Too shallow (VT10)?
</ol>
<hr />
<li> Working with short sequences -- when the scoring matrix matters -- (afternoon presentation)
<p>
<ol type='a'>
<li>Use
the <a href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=select&pgm=fx&query=295842262&db=p&annot_seq2=2&hide_align=1"  target='fasta_win'>FASTX
[pgm]</a> to compare the Honey bee GSTD1 mRNA (NM_001178028,
gi|GI:295842262) to the PIR1 database.  How does the sensitivity of
this translated DNA vs protein search compare with your earlier
protein:protein search?  Why might a translated DNA vs protein search be less sensitive than a protein:protein library search?  (Hint: how long is the mRNA?)
<p>
<li>
Now do the same search, but use only exon 3 of the Honey bee GSTD1 gene, which corresponds to nt 456-597.  Use the Subset-range to select the exon 3 nucleotides from GSTD1 and run <a href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=select&pgm=fx&query=295842262&db=p&annot_seq2=2&hide_align=1&ssr=456-597" target='fasta_win'>FASTX [pgm]</a> again.
<ol type='i'>
<li> What is the E()-value with of the most distantly related homolog with BLOSUM50?  What is the percent identity for this alignment?
<li> How long is the longest possible translated amino-acid sequence? How long is the protein alignment?
<li> Why do you think the search is so much less sensitive?
<li> What is the percent identity for the <tt>GSTA1_RAT</tt> alignment?
<li> Looking at percent identities in the Scoring matrix drop-down, what might be a better scoring matrix for the <tt>GSTA1_RAT</tt> alignment?
<li> What is the E()-value for that alignment with the new scoring matrix?
<li> What is the E()-value of the highest scoring non-homolog with that scoring matrix?
</ol>
<p>
<li>
Repeat the same translated FASTX search, using exon 2 of the Honey bee GSTD1 gene, which corresponds to nt 397-455 <a href="https://fasta.bioch.virginia.edu/fasta_www2/fasta_www.cgi?rm=select&pgm=fx&query=295842262&db=p&annot_seq2=2&hide_align=1&ssr=397-455"  target='fasta_win'>FASTX [pgm]</a>.
<ol type='i'>
<li>What is the best E()-value with BLOSUM50?  What is the percent identity for this alignment?
<li>How long is the longest possible translated amino-acid sequence? How long is the protein alignment?
<li>What would be a more appropriate scoring matrix based on the percent identity?
<li>What are the E()-values with that scoring matrix?
<li> What is the E()-value of the highest scoring non-homolog with that scoring matrix?
</ol>
<p>
</ol>
</ol>
<hr />
<p>
Where to get the FASTA package: 
<a href="https://github.com/wrpearson/fasta36/">github.com/wrpearson/fasta36<a>
<p>
The "normal" <a href="https://fasta.bioch.virginia.edu/">FASTA WWW site:</a>
<p>
Contact Bill Pearson: <a href="mailto:wrp@virginia.edu">wrp@virginia.edu</a>
<p>
<hr>
</body>
</html>
